Lib/test/test_ast.py

```diff
@@ -636,6 +636,21 @@ def test_issue39579_dotted_name_end_col_offset(self):
         attr_b = tree.body[0].decorator_list[0].value
         self.assertEqual(attr_b.end_col_offset, 4)
 
+    def test_ast_asdl_signature(self):
+        assert ast.withitem.__doc__ == "withitem(expr context_expr, expr? optional_vars)"
+        assert ast.GtE.__doc__ == "GtE"
+        assert ast.Name.__doc__ == "Name(identifier id, expr_context ctx)"
+        assert ast.cmpop.__doc__ == "cmpop = Eq | NotEq | Lt | LtE | Gt | GtE | Is | IsNot | In | NotIn"
+        expressions = []
+        for name, node in ast.__dict__.items():
+            if (
+                self._is_ast_node(name, node)
+                and issubclass(node, ast.expr)
+                and node.__doc__
+            ):
+                expressions.append(node.__doc__)
+        assert ast.expr.__doc__ == "expr = " + "\n       | ".join(expressions)
```

Comment:
Oh, I afraid it is a fragile test. There is no guarantee that classes are added in the `ast` module in the same order as they are enumerated in `ast.expr.__doc__`.  There is no guaranties that there are no subclasses of `ast.expr` subclasses. There is no guarantee that there are no aliases.

Maybe use `ast.expr.__subclasses__()` and compare unordered collections of lines with `astCountEqual`?

Reply:
Agreed. Will implement a better solution with count equal (also replace that asserts with assertEqual, sorry for that)